projects/storefrontlib/src/cms-components/product/product-interests/stock-notification/stock-notification.component.spec.ts

```diff
@@ -183,129 +154,108 @@ describe('StockNotificationComponent', () => {
+    getTestScheduler().flush();
 
-    const buttons = fixture.debugElement.queryAll(By.css('button'));
-    expect(buttons[0].nativeElement.textContent).toContain(
-      'stockNotification.notifyMe'
-    );
+    expect(ngbModal.open).not.toHaveBeenCalled();
+    expect(globalMessageService.add).toHaveBeenCalled();
   });
 
-  it('should show active channel link, when logged$ is true, subscribed$ is false, channelEnabled$ is false', () => {
-    authService.getUserToken.and.returnValue(of(userToken));
-    productInterestService.getBackInStockSubscribed.and.returnValue(of(false));
+  it('should show global message when delete stock notification success', () => {
+    productInterestService.getBackInStockSubscribed.and.returnValue(of(true));
     productInterestService.getDeleteBackInStockSuccess.and.returnValue(
-      of(true)
-    );
-    const emptyNotificationPreferenceList: BasicNotificationPreferenceList = {
-      preferences: [],
-    };
-    userService.getNotificationPreferences.and.returnValue(
-      of(emptyNotificationPreferenceList)
+      cold('-b|', { a: true })
     );
     fixture.detectChanges();
+    const button = el.query(By.css('[data-test="stocknotify-button"]'))
+      .nativeElement;
+    button.click();
 
-    component.logged$.subscribe(x => expect(x).toEqual(true));
-    component.subscribed$.subscribe(x => expect(x).toEqual(false));
-    component.channelEnabled$.subscribe(x => expect(x).toEqual(false));
-    const ahrefs = fixture.debugElement.queryAll(By.css('a'));
-    expect(ahrefs[0].nativeElement.textContent).toContain(
-      'stockNotification.channelsLink'
-    );
+    getTestScheduler().flush();
+
+    expect(ngbModal.open).not.toHaveBeenCalled();
+    expect(globalMessageService.add).toHaveBeenCalled();
   });
 
-  it('should show stop notify me, when logged$ is true, subscribed$ is true', () => {
-    authService.getUserToken.and.returnValue(of(userToken));
-    productInterestService.getBackInStockSubscribed.and.returnValue(of(true));
-    productInterestService.getDeleteBackInStockLoading.and.returnValue(
-      of(false)
-    );
+  it('should be able to show dialog and global message for create stock notification', () => {
     fixture.detectChanges();
 
-    component.logged$.subscribe(x => expect(x).toEqual(true));
-    component.subscribed$.subscribe(x => expect(x).toEqual(true));
-    component.unsubscribeLoading$.subscribe(x => expect(x).toEqual(false));
-    const buttons = fixture.debugElement.queryAll(By.css('button'));
-    expect(buttons[0].nativeElement.textContent).toContain(
-      'stockNotification.stopNotify'
-    );
+    const button = el.query(By.css('[data-test="stocknotify-button"]'))
+      .nativeElement;
+    button.click();
+    fixture.detectChanges();
+
+    expect(productInterestService.createBackInStock).toHaveBeenCalled();
+    expect(ngbModal.open).toHaveBeenCalled();
+    expect(globalMessageService.add).not.toHaveBeenCalled();
```

Comment:
Why the name is 'show dialog and global message' but you expect(globalMessageService.add).not.toHaveBeenCalled();  ?

Reply:
is there any message shown when subscribed the notification???

Reply:
we don't need to verify there is no message to shown